src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrLength = this.length;
+    let resultStr = '';
+
+    for (let i = 0; i < arrLength; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        resultStr += '';
+      } else {
+        resultStr += this[i];
+      }
```

Comment:
Do you really need to add a blank string? It's a redundant action. Try to invert your if statement and only add the thing that you need in your result